reference/programming-model.md

```diff
@@ -143,15 +144,84 @@ this.registerOutputs({
 })
 ```
 
+In addition to the usual resource options, components accept a set of [providers](#providers) to use for their child resources. If a component is itself a child of another component, its set of providers is inherited from its parent by default.
+
+```js
+let component = new MyResource("component", { providers: { aws: useast1, kubernetes: myk8s } });
+```
+
 For more information about components, see the [Pulumi Components](component-tutorial.html) tutorial.
 
+## Providers {#providers}
+
+A [CustomResource][pulumi.CustomResource] needs an associated resource provider to manage its Create, Read, Update, and Delete (_CRUD_) operations. This is in contrast to a [ComponentResource][pulumi.ComponentResource], whose logic is authored entirely in a Pulumi program's source language (e.g. Javascript or Python). By default, a `CustomResource`'s provider is determined based on its [package](#packages). This default provider is automatically created by Pulumi, and is configured using its package's [config values](#config). For example, the configuration and program below will create a single EC2 instance in the `us-west-2` region.
```

Comment:
Does `pulumi.CustomResource` correctly link to something?

Reply:
I'm not sure that `pulumi.CustomResource` or `pulumi.ComponentResource` link to anything, in fact. I moved this sentence from a different part of the reference; I'll check and fix these links if necessary.

Reply:
These do in fact link to the `@pulumi/pulumi` docs.